lib/Routing/DataObjectRoute.php

```diff
@@ -89,7 +89,7 @@ public function getContent(): ?object
     public function getRouteKey(): ?string
     {
         if ($this->object) {
-            return sprintf('data_object_%d_%s', $this->object->getId(), $this->getPath());
+            return sprintf('data_object_%d_%s_%s', $this->object->getId(), $this->getDefault('site'), $this->getPath());
```

Comment:
I think it would make sense to also add a `protected ?Site $site` property to this class, rather than using the default functionality from the parent. (see also `slug` and `object` property)